File: govern/data-security/ranger-2.3.0/agents-common/src/main/java/org/apache/ranger/plugin/model/UserInfo.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.apache.ranger.plugin.model;

import org.apache.ranger.plugin.util.RangerUserStoreUtil;
import org.codehaus.jackson.annotate.JsonAutoDetect;
import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.*;

@JsonAutoDetect(fieldVisibility=JsonAutoDetect.Visibility.ANY)
@JsonSerialize(include=JsonSerialize.Inclusion.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown=true)
@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class UserInfo extends RangerBaseModelObject implements java.io.Serializable {

    private static final long serialVersionUID = 1L;
    private String                  name;
    private String                  description;
    private Map<String, String>     otherAttributes;
    private Set<String>             groups;

    public UserInfo() {
        this(null, null, null);
    }

    public UserInfo(String name, String description, Map<String, String> otherAttributes) {
        setName(name);
        setDescription(description);
        setOtherAttributes(otherAttributes);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Map<String, String> getOtherAttributes() {
        return otherAttributes;
    }

    public void setOtherAttributes(Map<String, String> otherAttributes) {
        this.otherAttributes = otherAttributes == null ? new HashMap<>() : otherAttributes;
    }

    public Set<String> getGroups(){
        return this.groups;
    }

    public void setGroups(Set<String> groups){
        this.groups = groups;
    }

    @Override
    public String toString() {
        return "{name=" + name
                + ", description=" + description
                + ", otherAttributes=" + RangerUserStoreUtil.getPrintableOptions(otherAttributes)
                + ", groups=" + groups
                + "}";
    }
}

File: govern/data-security/ranger/security-admin/src/main/webapp/scripts/views/users/UserInfo.js
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

 
define(function(require){
    'use strict';

	var Backbone		= require('backbone');
	var Communicator	= require('communicator');
	var XAEnums 		= require('utils/XAEnums');
	var UserInfo_tmpl = require('hbs!tmpl/users/UserInfo_tmpl'); 
	
	var UserInfo = Backbone.Marionette.ItemView.extend(
	/** @lends UserInfo */
	{
		_viewName : 'UserInfo',
		
    	template: UserInfo_tmpl,
    	templateHelpers : function(){
    		return {
    			'groupList' : this.groupList,
    			'userList'	: this.userList,
    			'userModel'	: this.model.modelName == XAEnums.ClassTypes.CLASS_TYPE_XA_USER.modelName ? true : false,
    			'groupModel': this.model.modelName == XAEnums.ClassTypes.CLASS_TYPE_XA_GROUP.modelName ? true : false
    		};
    	},
        
    	/** ui selector cache */
    	ui: {
    		'showMoreBtn' : '[data-id="showMore"]',
    		'showLessBtn' : '[data-id="showLess"]'
    	},

		/** ui events hash */
		events: function() {
			var events = {};
			//events['change ' + this.ui.input]  = 'onInputChange';
			events['click ' + this.ui.showMoreBtn]  = 'onShowMoreClick';
			events['click ' + this.ui.showLessBtn]  = 'onShowLessClick';
			return events;
		},

    	/**
		* intialize a new UserInfo ItemView 
		* @constructs
		*/
		initialize: function(options) {
			console.log("initialized a UserInfo ItemView");

			_.extend(this, _.pick(options, 'groupList','userList'));
			
			this.bindEvents();
		},

		/** all events binding here */
		bindEvents : function(){
			/*this.listenTo(this.model, "change:foo", this.modelChanged, this);*/
			/*this.listenTo(communicator.vent,'someView:someEvent', this.someEventHandler, this)'*/
		},

		/** on render callback */
		onRender: function() {
			this.initializePlugins();
			if(this.ui.showMoreBtn.parent('td').find('span').length > 3){
				this.ui.showMoreBtn.show();	
				this.ui.showMoreBtn.parent('td').find('span').slice(3).hide();
			}
		},
		onShowMoreClick : function(){
			this.ui.showMoreBtn.parent('td').find('span').show();
			this.ui.showMoreBtn.hide();
			this.ui.showLessBtn.show();
		},
		onShowLessClick : function(){
			this.ui.showMoreBtn.parent('td').find('span').slice(3).hide();
			this.ui.showLessBtn.hide();
			this.ui.showMoreBtn.show();
		},

		/** all post render plugin initialization */
		initializePlugins: function(){
		},

		/** on close */
		onClose: function(){
		}

	});

	return UserInfo;
});
